docs/specs/build/resolve.yml

```diff
@@ -60,7 +92,38 @@ inputs:
   docs/dir/b.md: '<a target="_blank" href="c.md">c</a>'
   docs/dir/c.md:
 outputs:
-  docs/a.json: '{ "content": "<p>Link to <a target=\"_blank\" href=\"dir/c\">c</a></p>" }'
-  docs/dir/b.json: '{ "content": "<p><a target=\"_blank\" href=\"c\">c</a></p>" }'
+  docs/a.json: |
+    { "content": "<p>Link to <a target=\"_blank\" href=\"dir/c\">c</a></p>" }
+  docs/dir/b.json: |
+    { "content": "<p><a target=\"_blank\" href=\"c\">c</a></p>" }
   docs/dir/c.json:
   build.manifest:
+---
+# Show warning if the link is empty, missing or points to absolute file path
+inputs:
+  docfx.yml:
+  docs/a.md: |
+    [Empty]() [Invalid](a<>b.md) [!INCLUDE[](../b.md)]
+  b.md: |
+    [Missing](c.md)
+outputs:
+  docs/a.json: |
+    { "content": "<p><a href=\"\">Empty</a><a href=\"a%3C%3Eb.md\">Invalid</a><a href=\"c.md\">Missing</a></p>" }
+  build.manifest:
+  build.log: |
+    ["warning","link-is-empty","Link is empty","docs/a.md"]
```

Comment:
>empty [](start = 24, length = 5)

more than 1 error/warnings, will cause dynamically test failures

Reply:
For one file, the order is deterministic.

Reply:
but we need a way to tell the incoming developer that they should only verfy 1 error/warning or 1 file's error/warnings

---
In reply to: [193958853](https://github.com/dotnet/docfx/pull/2857#discussion_r193958853) [](ancestors = 193958853)

Reply:
it should be *few* and *stable*

---
In reply to: [193959072](https://github.com/dotnet/docfx/pull/2857#discussion_r193959072) [](ancestors = 193959072,193958853)